vyos/vyos-1x
# T5681: Firewall,Nat and Nat66: simplified and standarize interface ma…

## Change Summary

…tcher (valid for interfaces and groups) in firewal, nat and nat66.

(cherry picked from commit 51abbc0f1b2ccf4785cf7f29f1fe6f4af6007ee6)
## Types of changes

- [x] Other (please describe): Cherry-pick

## Related Task(s)

* https://vyos.dev/T5681

## Related PR(s)


## Component(s) name

firewall, nat
## Proposed changes


## How to test
VyOS configuration:
```
set firewall group interface-group IF01 interface 'bond99'
set firewall ipv4 forward filter rule 10 action 'accept'
set firewall ipv4 forward filter rule 10 inbound-interface name 'eth3'
set firewall ipv4 forward filter rule 10 outbound-interface group 'IF01'
set firewall ipv6 name FOO6 rule 199 action 'accept'
set firewall ipv6 name FOO6 rule 199 inbound-interface name '!vtun11'
set firewall ipv6 name FOO6 rule 199 outbound-interface group 'IF01'
set nat destination rule 10 inbound-interface name 'eth3'
set nat destination rule 10 translation address '192.0.2.1'
set nat source rule 20 outbound-interface group 'IF01'
set nat source rule 20 translation address 'masquerade'
set nat66 destination rule 10 inbound-interface name 'eth2'
set nat66 destination rule 10 translation address 'fe80::c4d3:3ff:feb3:98c2'
set nat66 source rule 10 outbound-interface name 'eth3'
set nat66 source rule 10 translation address 'masquerade'
```
Check:
```
vyos@r1# sudo nft -s list table ip vyos_nat | grep "oif\|iif"
		iifname "eth3" counter dnat to 192.0.2.1 comment "DST-NAT-10"
		oifname @I_IF01 counter masquerade comment "SRC-NAT-20"

vyos@r1# sudo nft -s list table ip6 vyos_nat | grep "oif\|iif"
		iifname "eth2" counter dnat to fe80::c4d3:3ff:feb3:98c2 comment "DST-NAT66-10"
		oifname "eth3" counter masquerade comment "SRC-NAT66-10"

vyos@r1# sudo nft -s list table ip vyos_filter | grep "oif\|iif"
		iifname "eth3" oifname @I_IF01 counter accept comment "FWD-filter-10"

vyos@r1# sudo nft -s list table ip6 vyos_filter | grep "oif\|iif"
		iifname != "vtun11" oifname @I_IF01 counter accept comment "NAM-FOO6-199"
```

## Smoketest result
```
vyos@r1:~$ /usr/libexec/vyos/tests/smoke/cli/test_firewall.py 
test_geoip (__main__.TestFirewall.test_geoip) ... ok
test_groups (__main__.TestFirewall.test_groups) ... ok
test_ipv4_advanced (__main__.TestFirewall.test_ipv4_advanced) ... ok
test_ipv4_basic_rules (__main__.TestFirewall.test_ipv4_basic_rules) ... ok
test_ipv4_mask (__main__.TestFirewall.test_ipv4_mask) ... ok
test_ipv4_state_and_status_rules (__main__.TestFirewall.test_ipv4_state_and_status_rules) ... ok
test_ipv6_advanced (__main__.TestFirewall.test_ipv6_advanced) ... ok
test_ipv6_basic_rules (__main__.TestFirewall.test_ipv6_basic_rules) ... ok
test_ipv6_mask (__main__.TestFirewall.test_ipv6_mask) ... ok
test_nested_groups (__main__.TestFirewall.test_nested_groups) ... ok
test_source_validation (__main__.TestFirewall.test_source_validation) ... ok
test_sysfs (__main__.TestFirewall.test_sysfs) ... ok
test_zone_basic (__main__.TestFirewall.test_zone_basic) ... ok

----------------------------------------------------------------------
Ran 13 tests in 41.489s

OK


vyos@r1:~$ /usr/libexec/vyos/tests/smoke/cli/test_nat.py
test_dnat (__main__.TestNAT.test_dnat) ... ok
test_dnat_negated_addresses (__main__.TestNAT.test_dnat_negated_addresses) ... ok
test_dnat_redirect (__main__.TestNAT.test_dnat_redirect) ... ok
test_dnat_without_translation_address (__main__.TestNAT.test_dnat_without_translation_address) ... ok
test_nat_balance (__main__.TestNAT.test_nat_balance) ... ok
test_nat_no_rules (__main__.TestNAT.test_nat_no_rules) ... ok
test_snat (__main__.TestNAT.test_snat) ... ok
test_snat_groups (__main__.TestNAT.test_snat_groups) ... ok
test_snat_required_translation_address (__main__.TestNAT.test_snat_required_translation_address) ... ok
test_static_nat (__main__.TestNAT.test_static_nat) ... ok

----------------------------------------------------------------------
Ran 10 tests in 24.059s

OK


vyos@r1:~$ /usr/libexec/vyos/tests/smoke/cli/test_nat66.py
test_destination_nat66 (__main__.TestNAT66.test_destination_nat66) ... ok
test_destination_nat66_prefix (__main__.TestNAT66.test_destination_nat66_prefix) ... ok
test_destination_nat66_protocol (__main__.TestNAT66.test_destination_nat66_protocol) ... ok
test_destination_nat66_without_translation_address (__main__.TestNAT66.test_destination_nat66_without_translation_address) ... ok
test_nat66_no_rules (__main__.TestNAT66.test_nat66_no_rules) ... ok
test_source_nat66 (__main__.TestNAT66.test_source_nat66) ... ok
test_source_nat66_address (__main__.TestNAT66.test_source_nat66_address) ... ok
test_source_nat66_protocol (__main__.TestNAT66.test_source_nat66_protocol) ... ok
test_source_nat66_required_translation_prefix (__main__.TestNAT66.test_source_nat66_required_translation_prefix) ... ok

----------------------------------------------------------------------
Ran 9 tests in 19.230s

OK

```

## Checklist:
